src/controllers/rendering/controllerrenderingscheduler.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <GL/gl.h>
+
+#include <QMutex>
+#include <QThread>
+#include <QWaitCondition>
+
+#include "controllers/legacycontrollermapping.h"
+#include "util/time.h"
+
+class Controller;
+class QByteArray;
+
+/// @brief This class is used to schedule rendering and handle dispatching of
+/// frame data for a screen controller. It is used by ControllerRenderingEngine
+/// to drive itself in the GUI thread.
+class ControllerRenderingScheduler : public QThread {
+    Q_OBJECT
+  public:
+    ControllerRenderingScheduler(const LegacyControllerMapping::ScreenInfo& screenInfo);
+
+    bool event(QEvent* event) override;
+
+  public slots:
+    void requestSend(Controller* controller, const QByteArray& frame);
+    void send(Controller* controller, const QByteArray& frame);
+
+  signals:
+    void frameRequested();
+    void sendRequested(Controller* controller, const QByteArray& frame);
```

Comment:
Well, whats the purpose of `requestSend` if it just re-emits `sendRequsted`, only to then in turn call `send` again?

Reply:
`requestSend` can be called by any threads, but it will make sure that the scheduler calls `send` in its own thread. This will be removed when I revert the single threaded version

Reply:
I see, I was hoping there was a simpler way to do this, but apparently not. In that case it would make sense to document how these member functions are intended to be used (one being threadsafe and the other one not). Or in that case it could even make sense to make the non-threadsafe one private.